### PR TITLE
[FEATURE] MoveAllowTableOnStandardPagesToTCAConfigurationRector

### PR DESCRIPTION
**Pull request by sabbelasichon**
Resolves: https://github.com/sabbelasichon/typo3-rector/issues/3260

### CODE REVIEW

File: rules/TYPO312/v0/MoveAllowTableOnStandardPagesToTCAConfigurationRector.php [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+     * @readonly
+     */
+    private IgnorePageTypeRestrictionRector $ignorePageTypeRestrictionRector;
+
+    /**
+     * @readonly
+     */
+    private FormatPerservingPrinter $formatPerservingPrinter;
+
+    private FilesystemInterface $filesystem;
```
Comment (simonschaufi) [inline rules/TYPO312/v0/MoveAllowTableOnStandardPagesToTCAConfigurationRector.php:54]: readonly missing

File: rules/TYPO312/v0/MoveAllowTableOnStandardPagesToTCAConfigurationRector.php [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    /**
+     * @param Expression $node
+     */
+    public function refactor(Node $node)
+    {
+        $staticMethodCall = $node->expr;
+
+        if (! $staticMethodCall instanceof Node\Expr\StaticCall) {
```
Comment (simonschaufi) [inline rules/TYPO312/v0/MoveAllowTableOnStandardPagesToTCAConfigurationRector.php:96]: import namespace completely

**Review by simonschaufi — CHANGES_REQUESTED** [review]

File: config/config.php [hunk 3]
```diff
@@ -23,4 +29,15 @@
     $rectorConfig->singleton(InjectMethodFactory::class);
     $rectorConfig->singleton(Typo3GlobalsFactory::class);
     $rectorConfig->singleton(Typo3NodeResolver::class);
+    $rectorConfig->bind(FilesystemInterface::class, static function () {
+        $argv = $_SERVER['argv'] ?? [];
+        $isDryRun = in_array('--dry-run', $argv);
```
Comment (simonschaufi) [inline config/config.php:34]: `-n` is also possible

File: config/v12/typo3-120.php [hunk 4]
```diff
@@ -160,4 +162,7 @@
     $rectorConfig->rule(MigrateContentObjectRendererLastTypoLinkPropertiesRector::class);
     $rectorConfig->rule(UsePageDoktypeRegistryRector::class);
     $rectorConfig->rule(UseServerRequestInsteadOfGeneralUtilityPostRector::class);
+    $rectorConfig->rule(\Ssch\TYPO3Rector\TYPO312\v0\MoveAllowTableOnStandardPagesToTCAConfigurationRector::class);
```
Comment (simonschaufi) [inline config/v12/typo3-120.php:165]: remove as duplicate to line 167

**Review by simonschaufi — CHANGES_REQUESTED** [review]

File: rules/TYPO312/v0/IgnorePageTypeRestrictionRector.php [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            'Move method ExtensionManagementUtility::allowTableOnStandardPages to TCA configuration',
+            [new ConfiguredCodeSample(<<<'CODE_SAMPLE'
+use TYPO3\CMS\Core\Utility\ExtensionManagementUtility;
+ExtensionManagementUtility::allowTableOnStandardPages('my_table');
+CODE_SAMPLE
+                , <<<'CODE_SAMPLE'
+$GLOBALS['TCA']['my_table']['ctrl']['security']['ignorePageTypeRestriction']
+CODE_SAMPLE
+                , [
+                    self::TABLE_CONFIGURATION => 'my_table',
```
Comment (simonschaufi) [inline rules/TYPO312/v0/IgnorePageTypeRestrictionRector.php:37]: can you explain why it is useful to configure this and maybe even necessary? why can only one table be defined and not multiple?

File: tests/Rector/v12/v0/MoveAllowTableOnStandardPagesToTCAConfigurationRector/config/configured_rule.php [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+declare(strict_types=1);
+
+use Rector\Config\RectorConfig;
+use Ssch\TYPO3Rector\TYPO312\v0\IgnorePageTypeRestrictionRector;
+use Ssch\TYPO3Rector\TYPO312\v0\MoveAllowTableOnStandardPagesToTCAConfigurationRector;
+
+return static function (RectorConfig $rectorConfig): void {
+    $rectorConfig->import(__DIR__ . '/../../../../../../config/config.php');
+    $rectorConfig->ruleWithConfiguration(IgnorePageTypeRestrictionRector::class, []);
```
Comment (simonschaufi) [inline tests/Rector/v12/v0/MoveAllowTableOnStandardPagesToTCAConfigurationRector/config/configured_rule.php:11]: how does the test actually work without a table configuration?

Comment (sabbelasichon) [reply tests/Rector/v12/v0/MoveAllowTableOnStandardPagesToTCAConfigurationRector/config/configured_rule.php]: I have removed IgnorePageTypeRestrictionRector. It is not in use anymore. It was before in my draft. Good catch

File: rules/TYPO312/v0/IgnorePageTypeRestrictionRector.php (hunk 5 above)
Comment (sabbelasichon) [reply rules/TYPO312/v0/IgnorePageTypeRestrictionRector.php]: I have removed IgnorePageTypeRestrictionRector. It is not in use anymore. It was before in my draft. Good catch

**Pull request merged**